ui/app/components/auth/splash-page.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+/**
+ * @module AuthRoutePage
+ * The Auth::SplashPage wraps OktaNumberChallenge and AuthForm to manage the login flow and is responsible for calling the authenticate method
+ *
+ * @example
+ * <Auth::SplashPage @namespaceQueryParam={{this.namespaceQueryParam}} @onAuthSuccess={{action "authSuccess"}} @oidcProviderQueryParam={{this.oidcProvider}} @cluster={{this.model}} @onNamespaceUpdate={{perform this.updateNamespace}} />
+ *
+ * @param {string} param - info about the param
+ * */
+
+export default class AuthSplashPage extends Component {
```

Comment:
All of these actions + getter are copy pasted from the `controllers/vault/cluster/auth` file